File: packages/engine/test/clock_test.dart
import 'package:engine/engine.dart';
import 'package:test/test.dart';

void main() {
  group('FakeClock', () {
    test('returns the point it was set to', () {
      final clock = Clock.fixed(DateTime(2026, 9, 6, 18, 30));
      expect(clock.now, DateTime(2026, 9, 6, 18, 30));
    });

    test('today strips the time of day', () {
      final clock = Clock.fixed(DateTime(2026, 9, 6, 18, 30));
      expect(clock.today, DateTime(2026, 9, 6));
    });

    test('advanceBy() moves forward in days', () {
      final clock = Clock.fixed(DateTime(2026, 9, 6)) as FakeClock;
      clock.advanceBy(days: 9);
      expect(clock.today, DateTime(2026, 9, 15));
    });
  });
}

File: infra/supabase/functions/_shared/planner/clock_test.ts
import { assertEquals } from './dev_deps.ts';
import { FakeClock } from './clock.ts';

Deno.test('FakeClock returns the point it was set to', () => {
  const clock = new FakeClock(new Date(2026, 8, 6, 18, 30));
  assertEquals(clock.now(), new Date(2026, 8, 6, 18, 30));
});

Deno.test('FakeClock today strips the time of day', () => {
  const clock = new FakeClock(new Date(2026, 8, 6, 18, 30));
  assertEquals(clock.today(), new Date(2026, 8, 6));
});

Deno.test('FakeClock advanceBy() moves forward in days', () => {
  const clock = new FakeClock(new Date(2026, 8, 6));
  clock.advanceBy({ days: 9 });
  assertEquals(clock.today(), new Date(2026, 8, 15));
});
